widget.subscribe.php

```diff
@@ -413,6 +415,38 @@ function add_subscriber() {
 				}
 			}
 
+            $recaptcha_token = sanitize_text_field( $_POST['captcha_token'] );
+
+			if ( ! empty( $sailthru['google_recaptcha_site_key'] ) && ! empty( $sailthru['google_recaptcha_secret'] ) && ! empty( $recaptcha_token ) ) {
+				write_log( "reCaptcha enabled, verifying" );
+				try {
+					$response = wp_remote_get( 'https://www.google.com/recaptcha/api/siteverify?secret=' . $sailthru['google_recaptcha_secret'] . '&response=' . $_POST['captcha_token'] );
+					$body = wp_remote_retrieve_body( $response );
+					$data = json_decode( $body, true );
+					if ( false == $data['success'] ) {
+						// failed, send an error message here, but keep it vague
+						$this->return_response(
+							array(
+								'success' => false,
+								'message' => 'Sorry, something went wrong and we could not subscribe you.'
+							)
+						);
+					}
+				} catch ( Exception $e ) {
+					write_log( $e->getMessage() );
```

Comment:
Does the message give us enough information to track down the exception here? Would writing the whole exception (`write_log( $e );`) make more sense?

Reply:
I think so considering we log the entire exception elsewhere.